kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java

```diff
@@ -339,6 +394,11 @@ public void assign(Set<AbstractKafkaPartitioner.PartitionMeta> assignment)
     return assignment;
   }
 
+  private boolean isIdempotent()
+  {
+    return windowDataManager != null && !(windowDataManager instanceof WindowDataManager.NoopWindowDataManager);
+  }
```

Comment:
Here, windowDataManager can't be null. By Default, the value of windowDataManager is  WindowDataManager.NoopWindowDataManager.